copy/copy.go

```diff
@@ -457,7 +634,11 @@ func (ic *imageCopier) copyUpdatedConfigAndManifest() ([]byte, error) {
 	}
 
 	ic.c.Printf("Writing manifest to image destination\n")
-	if err := ic.c.dest.PutManifest(manifest); err != nil {
+	if instanceDigest != nil {
+		tmpDigest := digest.FromBytes(manifest)
+		instanceDigest = &tmpDigest
+	}
+	if err := ic.c.dest.PutManifest(manifest, instanceDigest); err != nil {
```

Comment:
`manifest` here is being read from `pendingImage`, which should have had updates applied to it at this point.  Being able to call `manifest.Digest()` (which, yeah, we need to be consistent about that) means I need to rename the variable, though, so that's changed.  Using `MatchesDigest()` just forces us to compute the digest a second time if they don't match, though, so I don't see much of a gain from making that a second condition.

Reply:
> Using MatchesDigest() just forces us to compute the digest a second time if they don't match, though, so I don't see much of a gain from making that a second condition.

In _theory_ a `manifest.MatchesDigest()` check would allow preserving a digest using a non-default algorithm. In _practice_, yeah, `manifest.MatchesDigest` is just a `manifest.Digest` + a comparison, there isn’t currently a difference. Still, doing the theoretically right thing here would allow us to extend `manifest.MatchesDigest` in the future.

We can compensate for the repeated manifest digest computation by returning the computed manifest digest from this function and `copyOneImage`, and dropping
```go
		// Record the result of a possible conversion here.
		md, err := manifest.Digest(updatedManifest)
```
in `copyMultipleImages` ;-)